packages/contracts/src/PoolMetadataRegistry.sol

```diff
@@ -49,29 +48,39 @@ contract PoolMetadataRegistry is IPoolMetadataRegistry, PoolMetadataAuthorizatio
         }
     }
 
-    /// @notice Returns the authorizer contract for managing access control
-    /// @return The authorizer contract
-    function _getAuthorizer() internal view override returns (IAuthorizer) {
-        return _vault.getAuthorizer();
-    }
-
-    /// @notice Returns the owner of the pool that's currently being operated on
-    /// @return The owner of the pool
-    function _getOwner() internal view override returns (address) {
-        (address pool,) = _vault.getPool(_poolId);
-        return BasePoolAuthorization(pool).getOwner();
-    }
-
     /// @notice Updates the pool metadata CID
     /// @param poolId The pool ID to update the metadata
     /// @param metadataCID The metadataCID related to the new pool metadata
     function setPoolMetadata(bytes32 poolId, string memory metadataCID)
         public
         onlyRegisteredPool(poolId)
-        setPoolId(poolId)
-        authenticate
+        authenticate(poolId)
     {
         poolIdMetadataCIDMap[poolId] = metadataCID;
         emit PoolMetadataUpdated(poolId, metadataCID);
     }
+
+    /// @notice Computes the action ID for a given function selector.
+    /// @param selector The function selector to compute the action ID for.
+    /// @return The computed action ID.
+    function getActionId(bytes4 selector) internal view returns (bytes32) {
+        return keccak256(abi.encodePacked(bytes32(uint256(address(this))), selector));
+    }
+
+    /// @notice Determines whether an action can be performed by a given account
+    /// @param actionId The ID of the action to be performed
+    /// @param account The account to check for authorization
+    /// @return bool Returns true if the action can be performed, otherwise false
+    function _canPerform(bytes32 actionId, bytes32 poolId, address account) internal view returns (bool) {
+        (address pool,) = _vault.getPool(poolId);
+        address poolOwner = BasePoolAuthorization(pool).getOwner();
+
+        if (poolOwner != _DELEGATE_OWNER) {
+            // Only the owner can perform "owner only" actions, unless the owner is delegated.
+            return msg.sender == poolOwner;
+        } else {
+            // Non-owner actions are always processed via the Authorizer, as "owner only" ones are when delegated.
+            return _vault.getAuthorizer().canPerform(actionId, account, address(this));
```

Comment:
Should this be the pool address instead?

Reply:
Fixed, Thank you guys